modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -564,6 +568,18 @@ func (t TLSConfig) MakeTLSClientConfig(ctx caddy.Context) (*tls.Config, error) {
 		cfg.RootCAs = rootPool
 	}
 
+	if t.CARaw != nil {
+		caRaw, err := ctx.LoadModule(t, "CARaw")
+		if err != nil {
+			return nil, fmt.Errorf("failed to load ca module: %v", err)
+		}
+		ca, ok := caRaw.(caddytls.CA)
+		if !ok {
+			return nil, fmt.Errorf("CARaw module '%s' is not a certificate pool provider", ca)
+		}
+		cfg.ClientCAs = ca.CertPool()
```

Comment:
The `ClientCAs` field is for servers who are validating incoming client connections. In reverse-proxy transport, we're doing the opposite. We need the block above to be edited, i.e. modifying the source of root certificates.

Reply:
Sorry, forgot to change it from the code I ported from your PR. I fixed this. I also added a check to make sure we are not using the old and the new methods of getting the `RootCAs` at the same time. Thanks for the feedback.